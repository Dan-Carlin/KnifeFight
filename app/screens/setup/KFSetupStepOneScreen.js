import React from "react";
import { View, StyleSheet } from "react-native";

import colors from "../../config/colors";
import { ExitGraphic, NextGraphic } from "../../assets/buttons/navigation";
import GenerateNameButton from "../../components/GenerateNameButton";
import NavButton from "../../components/NavButton";
import Screen from "../../components/Screen";
import Text from "../../components/Text";
import TextInput from "../../components/TextInput";

const background = require("../../assets/backgrounds/kf_background_xxxhdpi.png");

function KFSetupStepOneScreen(props) {
  return (
    <Screen style={styles.screenContainer} background={background}>
      <View style={styles.topContainer} />
      <View style={styles.bodyContainer} behavior="padding">
        <Text style={styles.title}>First...</Text>
        <Text style={styles.text}>
          Let's come up with a name for your gang.
        </Text>
        <TextInput
          style={styles.textInput}
          placeholder="Name your gang"
          maxLength={22}
        />
        <View style={styles.nameButton}>
          <GenerateNameButton style={styles.button} />
        </View>
        <View style={styles.navContainer}>
          <NavButton
            style={styles.navButton}
            Graphic={ExitGraphic}
            onPress={() => console.log("Exit button pressed.")}
          />
          <NavButton
            style={styles.navButton}
            Graphic={NextGraphic}
            onPress={() => console.log("Next button pressed.")}
          />
        </View>
      </View>
      <View style={styles.bottomContainer} />
    </Screen>
  );
}

const styles = StyleSheet.create({
  bodyContainer: {
    alignItems: "center",
    flex: 0.4,
    justifyContent: "center",
    paddingHorizontal: 25,
  },
  bottomContainer: {
    alignItems: "center",
    flex: 0.3,
  },
  nameButton: {
    height: 40,
    marginBottom: 15,
    width: "85%",
  },
  navButton: {
    height: "100%",
    width: "100%",
  },
  navContainer: {
    flexDirection: "row",
    height: 90,
  },
  screenContainer: {
    flex: 1,
  },
  text: {
    textAlign: "center",
  },
  textInput: {
    color: colors.white,
    fontSize: 20,
    marginHorizontal: 20,
    textAlign: "center",
  },
  title: {
    fontFamily: "default",
    fontSize: 50,
    textAlign: "center",
  },
  topContainer: {
    alignItems: "center",
    flex: 0.3,
  },
});

export default KFSetupStepOneScreen;
