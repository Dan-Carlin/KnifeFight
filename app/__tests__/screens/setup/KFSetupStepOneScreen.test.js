import React from "react";
import { fireEvent, render } from "@testing-library/react-native";
import renderer from "react-test-renderer";

import KFSetupStepOneScreen from "../../../screens/setup/KFSetupStepOneScreen";

describe("Setup first screen tests", () => {
  it("renders first Setup screen correctly", () => {
    const tree = renderer.create(<KFSetupStepOneScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // generates a random name
  // it("generates a random name", () => {
  //   const { getByTestId, getByPlaceholderText, get } = render(
  //     <KFSetupStepOneScreen />
  //   );
  //   const button = getByTestId("btn_generateName");
  //   const textInput = getByPlaceholderText(/Name your gang/i);

  //   fireEvent.press(button);

  //   expect(textInput.value).toBeDefined();
  // });

  // activates Next button when name field is not empty

  // stores name value when next button is tapped
});
