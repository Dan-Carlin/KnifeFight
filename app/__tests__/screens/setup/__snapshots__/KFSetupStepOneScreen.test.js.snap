// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Setup first screen tests renders first Setup screen correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Image
    resizeMode="cover"
    source={1}
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Array [
        Object {
          "paddingTop": 20,
        },
        Object {
          "flex": 1,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 0.3,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 0.4,
          "justifyContent": "center",
          "paddingHorizontal": 25,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontFamily": "Avenir",
              "fontSize": 16,
              "width": "100%",
            },
            Object {
              "fontFamily": "default",
              "fontSize": 50,
              "textAlign": "center",
            },
          ]
        }
      >
        First...
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontFamily": "Avenir",
              "fontSize": 16,
              "width": "100%",
            },
            Object {
              "textAlign": "center",
            },
          ]
        }
      >
        Let's come up with a name for your gang.
      </Text>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#414042",
              "borderColor": "#bcbec0",
              "borderRadius": 30,
              "borderWidth": 2,
              "marginVertical": 10,
              "padding": 15,
            },
            Object {
              "width": "100%",
            },
          ]
        }
      >
        <TextInput
          allowFontScaling={true}
          maxLength={22}
          placeholder="Name your gang"
          placeholderTextColor="#7a7b7c"
          rejectResponderTermination={true}
          style={
            Object {
              "color": "#fff",
              "fontSize": 20,
              "marginHorizontal": 20,
              "textAlign": "center",
            }
          }
          underlineColorAndroid="transparent"
        />
      </View>
      <View
        style={
          Object {
            "height": 40,
            "marginBottom": 15,
            "width": "85%",
          }
        }
      >
        <SvgMock
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        />
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "height": 90,
            "justifyContent": "space-between",
            "width": "90%",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "height": "100%",
              "opacity": 1,
              "width": "45%",
            }
          }
        >
          <SvgMock />
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "height": "100%",
              "opacity": 1,
              "width": "45%",
            }
          }
        >
          <SvgMock />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 0.3,
        }
      }
    />
  </RCTSafeAreaView>
</View>
`;
